In [162]:
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                import streamlit as st

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                import pickle

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                import numpy as np
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                import pandas as pd 
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                from numbers import Number
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                from scipy import stats

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                import matplotlib.pyplot as plt
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                import seaborn as sns

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                import itertools
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                import statsmodels.api as sm
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                from sklearn.preprocessing import OrdinalEncoder
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                from sklearn.linear_model import LinearRegression
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                from sklearn.model_selection import train_test_split
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                from sklearn.metrics import mean_absolute_error
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                from sklearn.metrics import mean_squared_error
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                from sklearn.preprocessing import StandardScaler
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                from sklearn.preprocessing import OneHotEncoder
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                from sklearn.preprocessing import PolynomialFeatures

In [163]:
st.title("Kings County Housing Dashboard")

st.header("Interactive Housing Predictive Model Widget")

st.subheader("Map of Kings County, WA Housing Data")

DeltaGenerator()

In [212]:
df = pickle.load(open('df.pkl','rb'))
model = pickle.load(open('model.pkl','rb'))
ss = pickle.load(open('scaler.pkl','rb'))

In [198]:
list(df.loc[:5].columns)

['bedrooms',
 'bathrooms',
 'sqft_living',
 'sqft_lot',
 'floors',
 'sqft_above',
 'sqft_basement',
 'sqft_patio',
 'yr_built',
 'yr_renovated',
 'lat',
 'long',
 'distance_to_amazon',
 'grade',
 'condition',
 'view',
 'waterfront_NO',
 'waterfront_YES',
 'greenbelt_NO',
 'greenbelt_YES',
 'nuisance_NO',
 'nuisance_YES',
 'heat_source_Electricity',
 'heat_source_Electricity/Solar',
 'heat_source_Gas',
 'heat_source_Gas/Solar',
 'heat_source_Oil',
 'heat_source_Oil/Solar',
 'heat_source_Other',
 'sewer_system_PRIVATE',
 'sewer_system_PRIVATE RESTRICTED',
 'sewer_system_PUBLIC',
 'sewer_system_PUBLIC RESTRICTED',
 'zip_code_98001',
 'zip_code_98002',
 'zip_code_98003',
 'zip_code_98004',
 'zip_code_98005',
 'zip_code_98006',
 'zip_code_98007',
 'zip_code_98008',
 'zip_code_98010',
 'zip_code_98011',
 'zip_code_98014',
 'zip_code_98019',
 'zip_code_98022',
 'zip_code_98023',
 'zip_code_98024',
 'zip_code_98027',
 'zip_code_98028',
 'zip_code_98029',
 'zip_code_98030',
 'zip_code_98031',
 

In [254]:
house = df.loc[:0, :]

In [221]:
col = df.columns

In [229]:
df

bedrooms  bathrooms  sqft_living  sqft_lot  floors  sqft_above  \
0             4        1.0         1180      7140     1.0        1180   
1             5        2.5         2770      6703     1.0        1570   
2             6        2.0         2880      6156     1.0        1580   
3             3        3.0         2160      1400     2.0        1090   
4             2        2.0         1120       758     2.0        1120   
...         ...        ...          ...       ...     ...         ...   
29144         5        2.0         1910      4000     1.5        1600   
29145         3        2.0         2020      5800     2.0        2020   
29146         3        2.0         1620      3600     1.0         940   
29147         3        2.5         2570      2889     2.0        1830   
29148         3        1.5         1200     11058     1.0        1200   

       sqft_basement  sqft_patio  yr_built  yr_renovated  ...  zip_code_98155  \
0                  0          40      1969             0  ...             0.0   
1               1570         240      1950             0  ...             0.0   
2               1580           0      1956             0  ...             0.0   
3               1070         270      2010             0  ...             0.0   
4                550          30      2012             0  ...             0.0   
...              ...         ...       ...           ...  ...             ...   
29144           1130         210      1921             0  ...             0.0   
29145              0         520      2011             0  ...             0.0   
29146            920         110      1995             0  ...             0.0   
29147            740         100      2006             0  ...             0.0   
29148              0           0      1965             0  ...             0.0   

       zip_code_98166  zip_code_98168  zip_code_98177  zip_code_98178  \
0                 0.0             0.0             0.0             0.0   
1                 0.0             0.0             0.0             0.0   
2                 0.0             0.0             0.0             1.0   
3                 0.0             0.0             0.0             0.0   
4                 0.0             0.0             0.0             0.0   
...               ...             ...             ...             ...   
29144             0.0             0.0             0.0             0.0   
29145             0.0             0.0             0.0             0.0   
29146             0.0             0.0             0.0             0.0   
29147             0.0             0.0             0.0             0.0   
29148             0.0             0.0             0.0             0.0   

       zip_code_98188  zip_code_98198  zip_code_98199  zip_code_98224  \
0                 0.0             0.0             0.0             0.0   
1                 0.0             0.0             0.0             0.0   
2                 0.0             0.0             0.0             0.0   
3                 0.0             0.0             0.0             0.0   
4                 0.0             0.0             0.0             0.0   
...               ...             ...             ...             ...   
29144             0.0             0.0             0.0             0.0   
29145             0.0             0.0             0.0             0.0   
29146             0.0             0.0             0.0             0.0   
29147             0.0             0.0             0.0             0.0   
29148             0.0             0.0             0.0             0.0   

       zip_code_98288  
0                 0.0  
1                 0.0  
2                 0.0  
3                 0.0  
4                 0.0  
...               ...  
29144             0.0  
29145             0.0  
29146             0.0  
29147             0.0  
29148             0.0  

[29149 rows x 109 columns]

In [243]:
row['sqft_living'] = 1990.0

/var/folders/vd/6t9vjg9j7t9dfc844730ntn80000gn/T/ipykernel_6017/3991693033.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  row['sqft_living'] = 1990.0


In [244]:
row

bedrooms             4.0
bathrooms            3.0
sqft_living       1990.0
sqft_lot          5688.0
floors               1.0
                   ...  
zip_code_98188       0.0
zip_code_98198       0.0
zip_code_98199       0.0
zip_code_98224       0.0
zip_code_98288       0.0
Name: 5, Length: 109, dtype: float64

In [222]:
row = df.loc[5]

In [245]:
test_2 = pd.DataFrame(row, col).T

In [246]:
stand_2 = ss.transform(test_2)

In [247]:
model.predict(stand_2)

array([635715.80555622])

In [211]:
test

bedrooms  bathrooms  sqft_living  sqft_lot  floors  sqft_above  \
0         4        1.0         1180      7140     1.0        1180   

   sqft_basement  sqft_patio  yr_built  yr_renovated  ...  zip_code_98155  \
0              0          40      1969             0  ...             0.0   

   zip_code_98166  zip_code_98168  zip_code_98177  zip_code_98178  \
0             0.0             0.0             0.0             0.0   

   zip_code_98188  zip_code_98198  zip_code_98199  zip_code_98224  \
0             0.0             0.0             0.0             0.0   

   zip_code_98288  
0             0.0  

[1 rows x 109 columns]

In [208]:
a = stand.reshape(1,-1)

In [210]:
stand = ss.transform(test)

In [209]:
model.predict(a)

array([336579.80555622])

In [188]:
mode_house = modeled_df.iloc[0:1,:]

In [123]:
mode_house.loc[:,'yr_built'] = 2021

/var/folders/vd/6t9vjg9j7t9dfc844730ntn80000gn/T/ipykernel_6017/2844118229.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  mode_house.loc[:,'yr_built'] = 2021


In [125]:
standard_mode_house = ss.transform(mode_house)

In [126]:
result_mode_house = model.predict(standard_mode_house)[0]

In [127]:
result_mode_house

335011.8055562172

In [40]:
st.write('Median Price for Kings County:', '$869,000')

In [128]:
kc_zips = [98001, 98002,98003, 98004,98005,98006, 98007,98008,98009, 98010, 98011, 98013, 98014,98015,98019,98022,98023,98024,98025,98027,98028,98029,98030,98031,98032,98033,98034,98035,98038,98039,98040,98041,98042,98045,98047,98050,98051,98052,98053,98054,98055,98056,98057,98058,98059,98062,98063,98064,98065,98070,98071,98072,98073,98074,98075,98077,98083,98089,98092,98093,98101,98102,98103,98104,98105,98106,98107,98108,98109,98111,98112,98113,98114,98115,98116,98117,98118,98119,98121,98122,98124,98125,98126,98127,98129,98131,98132,98133,98134,98136,98138,98139,98141,98144,98145,98146,98148,98151,98154,98155,98158,98160,98161,98164,98165,98166,98168,98170,98171,98174,98175,98177,98178,98181,98184,98185,98188,98190,98191,98194,98195,98198,98199,98224,98288]


In [255]:
house

bedrooms  bathrooms  sqft_living  sqft_lot  floors  sqft_above  \
0         4        1.0         1180      7140     1.0        1180   

   sqft_basement  sqft_patio  yr_built  yr_renovated  ...  zip_code_98155  \
0              0          40      1969             0  ...             0.0   

   zip_code_98166  zip_code_98168  zip_code_98177  zip_code_98178  \
0             0.0             0.0             0.0             0.0   

   zip_code_98188  zip_code_98198  zip_code_98199  zip_code_98224  \
0             0.0             0.0             0.0             0.0   

   zip_code_98288  
0             0.0  

[1 rows x 109 columns]

In [270]:
new_house['zip_code_98007']

0    0.0
Name: zip_code_98007, dtype: float64

new_house.columns[-43]

In [299]:
list(house.columns)

['bedrooms',
 'bathrooms',
 'sqft_living',
 'sqft_lot',
 'floors',
 'sqft_above',
 'sqft_basement',
 'sqft_patio',
 'yr_built',
 'yr_renovated',
 'lat',
 'long',
 'distance_to_amazon',
 'grade',
 'condition',
 'view',
 'waterfront_NO',
 'waterfront_YES',
 'greenbelt_NO',
 'greenbelt_YES',
 'nuisance_NO',
 'nuisance_YES',
 'heat_source_Electricity',
 'heat_source_Electricity/Solar',
 'heat_source_Gas',
 'heat_source_Gas/Solar',
 'heat_source_Oil',
 'heat_source_Oil/Solar',
 'heat_source_Other',
 'sewer_system_PRIVATE',
 'sewer_system_PRIVATE RESTRICTED',
 'sewer_system_PUBLIC',
 'sewer_system_PUBLIC RESTRICTED',
 'zip_code_98001',
 'zip_code_98002',
 'zip_code_98003',
 'zip_code_98004',
 'zip_code_98005',
 'zip_code_98006',
 'zip_code_98007',
 'zip_code_98008',
 'zip_code_98010',
 'zip_code_98011',
 'zip_code_98014',
 'zip_code_98019',
 'zip_code_98022',
 'zip_code_98023',
 'zip_code_98024',
 'zip_code_98027',
 'zip_code_98028',
 'zip_code_98029',
 'zip_code_98030',
 'zip_code_98031',
 

In [301]:
st.image('house_pic.jpg')
    
new_house


beds = st.selectbox('Select Number of Bedrooms',[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 13])

baths = st.selectbox('Select Number of Bathrooms',[0.0,0.5,1.0,1.5,2.0,2.5,3.0,3.5,4.0,4.5,5.0
                                                  ,5.5,6.0, 6.5,7.0,7.5,8.0,8.5,9.5
                                                  ,10.0,10.5])

zip_code = st.selectbox('Select a Zip Code', kc_zips)
zip_col = 'zip_code_' + str(zip_code)

sqft_living = st.number_input('Input Square Feet of Living Space')


new_house = new_house.replace({'bedrooms': list(new_house['bedrooms'])}, beds)
new_house = new_house.replace({'bathrooms': list(new_house['bathrooms'])}, baths)
new_house = new_house.replace({zip_col: list(new_house[zip_col])}, 1.0)
new_house = new_house.replace({'sqft_living': list(new_house['sqft_living'])}, sqft_living)



st.number_input

st.text_input


standard_new_house = ss.transform(new_house)
    
result_new_house = model.predict(standard_new_house)[0]

if st.button('View House Sale Price'):
    st.write('This House is valued at:', result_new_house)

In [305]:
og_house = new_house

In [307]:
new_house

bedrooms  bathrooms  sqft_living  sqft_lot  floors  sqft_above  \
0         0        0.0            0      7140     1.0        1180   

   sqft_basement  sqft_patio  yr_built  yr_renovated  ...  zip_code_98155  \
0              0          40      1969             0  ...             0.0   

   zip_code_98166  zip_code_98168  zip_code_98177  zip_code_98178  \
0             0.0             0.0             0.0             0.0   

   zip_code_98188  zip_code_98198  zip_code_98199  zip_code_98224  \
0             0.0             0.0             0.0             0.0   

   zip_code_98288  
0             0.0  

[1 rows x 109 columns]

beds = 7
baths = 5
zip_col = 98198
zip_col = 'zip_code_' + str(zip_code)
sqft_living = 1180

new_house = new_house.replace({'bedrooms': list(new_house['bedrooms'])}, beds)
new_house = new_house.replace({'bathrooms': list(new_house['bathrooms'])}, baths)
new_house = new_house.replace({zip_col: list(new_house[zip_col])}, 1.0)
new_house = new_house.replace({'sqft_living': list(new_house['sqft_living'])}, sqft_living)


standard_new_house = ss.transform(new_house)
    
result_new_house = model.predict(standard_new_house)[0]

result_new_house

standard_new_house = ss.transform(test)
    
result_new_house = model.predict(standard_new_house)[0]
result_new_house